pkg/kallsyms/kallsyms_test.go

```diff
@@ -198,29 +237,147 @@ func TestRealKAllSymsParsing(t *testing.T) {
 	}
 }
 
-// BenchmarkKernelSymbolAddressJITParsing searches for the symbol while
-// reading the kallsyms file
-func BenchmarkKernelSymbolAddressJITParsing(b *testing.B) {
-	utilstest.RequireRoot(b)
-	b.ReportAllocs()
+func TestNewKAllSymsFromReaderErrors(t *testing.T) {
+	tests := []struct {
+		name        string
+		input       string
+		expectedErr string
+	}{
+		{
+			name:        "invalid_format_less_fields",
+			input:       "0000000000000000 A",
+			expectedErr: "line \"0000000000000000 A\" has less than 3 fields",
+		},
+		{
+			name:        "invalid_address",
+			input:       "INVALID_ADDR A fixed_percpu_data",
+			expectedErr: "strconv.ParseUint",
+		},
+		{
+			name:        "malformed_line_no_fields",
+			input:       "0000000000000000",
+			expectedErr: "line \"0000000000000000\" has less than 3 fields",
+		},
+	}
 
-	for n := 0; n < b.N; n++ {
-		exists := SymbolExists("security_bprm_check")
-		require.Equal(b, true, exists, "exists")
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			reader := strings.NewReader(tt.input)
+			_, err := NewKAllSymsFromReader(reader)
+			require.Error(t, err)
+			require.Contains(t, err.Error(), tt.expectedErr)
+		})
 	}
 }
 
-// BenchmarkKernelSymbolAddressPreLoading loads the full kallsyms before
-// looking up the symbol
-func BenchmarkKernelSymbolAddressPreLoading(b *testing.B) {
```

Comment:
This function is used for benchmarking (see `b *testing.B`). This function shouldn't be removed